src/vs/workbench/contrib/customEditor/browser/customEditors.ts

```diff
@@ -135,20 +136,23 @@ export class CustomEditorService extends Disposable implements ICustomEditorServ
 	}
 
 	private createDiffEditorInput(
-		editor: DiffEditorInput,
+		editor: IResourceDiffEditorInput,
 		editorID: string,
 		group: IEditorGroup
 	): DiffEditorInput {
-		const createEditorForSubInput = (subInput: IEditorInput, editorID: string, customClasses: string): EditorInput | undefined => {
+		const createEditorForSubInput = (subInput: IResourceEditorInput | IUntitledTextResourceEditorInput, editorID: string, customClasses: string): EditorInput | undefined => {
+			if (!subInput.resource) {
+				return;
+			}
 			// We check before calling this call back that both resources are defined
-			const input = CustomEditorInput.create(this.instantiationService, subInput.resource!, editorID, group.id, { customClasses });
+			const input = CustomEditorInput.create(this.instantiationService, subInput.resource, editorID, group.id, { customClasses });
 			return input instanceof EditorInput ? input : undefined;
 		};
 
-		const modifiedOverride = createEditorForSubInput(editor.modifiedInput, editorID, 'modified');
-		const originalOverride = createEditorForSubInput(editor.originalInput, editorID, 'original');
+		const modifiedOverride = createEditorForSubInput(editor.modifiedInput, editorID, 'modified') ?? this.editorService.createEditorInput(editor.modifiedInput);
```

Comment:
@bpasero This is why I needed to change createEditorInput to return an `EditorInput`. Since every implementation was returning one anyways I didn't feel it was a huge change.